### PR TITLE
Add snippets for julia and delete pi and phi

### PR DESCRIPTION
**Pull request by benneti**
Removing phi and pi as it can cause conflicts and problems in julia and latex modes (also in org-mode) and other programming modes, see https://github.com/hlissner/doom-snippets/issues/39.
Also add snippets for julia-mode, some snippets where taken from https://github.com/AndreaCrotti/yasnippet-snippets, I left the contributors name in these snippet files.

### CODE REVIEW

File: julia-mode/dfun [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- mode: snippet -*-
+# name: @doc ... function ... end
+# key: dfun
+# uuid: dfun
+# condition: t
```
Comment (hlissner) [inline julia-mode/dfun:5]: Remove redundant condition

File: julia-mode/.yas-setup.el [hunk 2]
```diff
@@ -0,0 +1,12 @@
+;;; ~/Repos/bennetis_dotfiles/emacs/.doom.d/snippets/julia-mode/.yas-setup.el -*- lexical-binding: t; -*-
+
+
+
+(require 'yasnippet)
```
Comment (hlissner) [inline julia-mode/.yas-setup.el:5]: Remove redundant `require`. Yasnippet is loaded if this file is being read.

File: julia-mode/.yas-setup.el [hunk 3]
```diff
@@ -0,0 +1,12 @@
+;;; ~/Repos/bennetis_dotfiles/emacs/.doom.d/snippets/julia-mode/.yas-setup.el -*- lexical-binding: t; -*-
```
Comment (hlissner) [inline julia-mode/.yas-setup.el:1]: Replace personal path with `julia-mode/.yas-setup.el`

File: julia-mode/.yas-setup.el (hunk 2 above)
Comment (hlissner) [reply julia-mode/.yas-setup.el:5]: And ensure there is only one blank newline after the filename comment header.

File: julia-mode/begin [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: begin ... end
+# key: begin
+# --
+begin
+    $0
```
Comment (hlissner) [inline julia-mode/begin:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/dfun [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+${3:Documentation of function.}
+
+# Examples
+\`\`\`jldoctest
+julia> $1($4)
+insert result of $1($4)
+\`\`\`
+"""
+function ${1:name}(${2:args})
+    $0
```
Comment (hlissner) [inline julia-mode/dfun:20]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/do [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# -*- mode: snippet -*-
+# name: do ... ... end
+# key: do
+# contributor: hankail05
+# --
+do ${1:x}
+    $0
```
Comment (hlissner) [inline julia-mode/do:7]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/for [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: for ... ... end
+# key: for
+# --
+for ${1:i} in ${3:1:n}
+    $0
```
Comment (hlissner) [inline julia-mode/for:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/for [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: for ... ... end
+# key: for
+# --
+for ${1:i} in ${3:1:n}
```
Comment (hlissner) [inline julia-mode/for:5]: Why is the second tab stop numbered `3`? Should it not be `2`?

File: julia-mode/fun [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: function(...) ... end
+# key: fun
+# --
+function ${1:name}(${2:args})
+    $0
```
Comment (hlissner) [inline julia-mode/fun:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/if [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: if ... ... end
+# key: if
+# --
+if ${1:cond}
+    $0
```
Comment (hlissner) [inline julia-mode/if:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/ife [hunk 11]
```diff
@@ -0,0 +1,10 @@
+# -*- mode: snippet -*-
+# name: if ... ... else ... end
+# key: ife
+# --
+if ${1:cond}
+    ${2:true}
```
Comment (hlissner) [inline julia-mode/ife:6]: Please add \`%\` before `${2:true}` to give this snippet implicit selection support.

```diff
-  ${2:true}
+  `%`${2:true}
```

File: julia-mode/let [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: let ... ... end
+# key: let
+# --
+let ${1:binding}
+    $0
```
Comment (hlissner) [inline julia-mode/let:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/macro [hunk 13]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: macro(...) ... end
+# key: macro
+# --
+macro ${1:macro}(${2:args})
+    $0
```
Comment (hlissner) [inline julia-mode/macro:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/module [hunk 14]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: module ... ... end
+# key: module
+# --
+module ${1:name}
+$0
```
Comment (hlissner) [inline julia-mode/module:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/mutstr [hunk 15]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: mutable struct ... end
+# key: mutstr
+# --
+mutable struct ${1:name}
+    $0
```
Comment (hlissner) [inline julia-mode/mutstr:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/quote [hunk 16]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: quote ... end
+# key: quote
+# --
+quote
+    $0
```
Comment (hlissner) [inline julia-mode/quote:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/struct [hunk 17]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: struct ... end
+# key: struct
+# --
+struct ${1:name}
+    $0
```
Comment (hlissner) [inline julia-mode/struct:6]: Please add \`%\` before `$0` to give this snippet implicit selection support.

```diff
-  $0
+  `%`$0
```

File: julia-mode/try [hunk 18]
```diff
@@ -0,0 +1,11 @@
+# -*- mode: snippet -*-
+# name: try ... catch ... end
+# key: try
+# contributor: hankail05
+# --
+try
+    ${1:expr}
```
Comment (hlissner) [inline julia-mode/try:7]: Please add \`%\` before `${1:expr}` to give this snippet implicit selection support.

```diff
-  ${1:expr}
+  `%`${1:expr}
```

File: julia-mode/tryf [hunk 19]
```diff
@@ -0,0 +1,13 @@
+# -*- mode: snippet -*-
+# name: try ... catch ... finally ... end
+# key: try
+# contributor: hankail05
+# --
+try
+    ${1:expr}
```
Comment (hlissner) [inline julia-mode/tryf:7]: Please add \`%\` before `${1:expr}` to give this snippet implicit selection support.

```diff
-  ${1:expr}
+  `%`${1:expr}
```

File: julia-mode/while [hunk 20]
```diff
@@ -0,0 +1,9 @@
+# -*- mode: snippet -*-
+# name: while ... ... end
+# key: while
+# contributor: hankail05
+# --
+while ${1:cond}
+    ${2:body}
```
Comment (hlissner) [inline julia-mode/while:7]: Please add \`%\` before `${2:body}` to give this snippet implicit selection support.

```diff
-  ${2:body}
+  `%`${2:body}
```

**Comment by benneti** [comment]
implemented the requested changes. Before you merge could you tell me what this implicit selection does? I did not find it https://joaotavora.github.io/yasnippet/snippet-development.html nor by a quick search in doom emacs.
Also I am not completely sure how YASnippet works so would it be better to insert blocks such that $0 is after the block or like how I did it right now?

**Comment by hlissner** [comment]
Quoting the relevant section of yasnippet's docs

> From version 0.6, snippets expansions are run with some special Emacs-lisp variables bound. One of this is yas-selected-text. You can therefore define a snippet like:
>
> for ($1;$2;$3) {
>          `yas-selected-text`$0
>       }
>
> to "wrap" the selected region inside your recently inserted snippet. Alternatively, you can also customize the variable yas-wrap-around-region to t which will do this automatically. 

This snippets library aliases `yas-selected-text` to `%` [here](https://github.com/hlissner/doom-snippets/blob/master/doom-snippets-lib.el#L9) for convenience.

Hope that helps!

**Comment by hlissner** [comment]
Also: there are two unresolved comments in the last review. See above.

**Comment by benneti** [comment]
Ups, forgot to push this change and thanks for the reference.

**Comment by hlissner** [comment]
No problem! Thanks for the help!

**Pull request merged**